### ISSUE TITLE
[REQ] add response-only chat and ShareGPT support

### ISSUE THREAD

**Issue by sozercan** [kaito-project/aikit#823]
Parent: #214

Depends on #598 and #822.

## Summary

Add explicit response-only SFT for `type: messages` and add `type: sharegpt` as a deterministic compatibility adapter.

```yaml
datasets:
  - source: acme/conversations
    type: messages

config:
  unsloth:
    loss: response
```

ShareGPT input:

```json
{
  "conversations": [
    {"from": "human", "value": "What is 2+2?"},
    {"from": "gpt", "value": "4"}
  ]
}
```

## Motivation

Native `assistant_only_loss` is not a reliable default for the frozen stack because it requires tokenizer templates that emit assistant masks, generally through `{% generation %}` regions. Tested Qwen and Llama templates did not reliably provide those masks.

Unsloth's `train_on_responses_only` is the practical chat response-only path, provided marker detection and resulting labels are validated. Unsloth's generic ShareGPT standardizer is heuristic and can fail on valid one-row or metadata-rich datasets, so AIKit should own an explicit adapter.

## Proposed changes

- Add an SFT loss option:
  - `all` — existing full-sequence chat behavior.
  - `response` — supported initially for `messages` and `sharegpt`.
- Keep `all` as the default for chat datasets.
- Reuse canonical message validation and chat-template rendering from #822.
- For `loss: response`:
  - Apply Unsloth response-only masking after rendering.
  - Validate that response markers can be derived.
  - Fail if any prepared training set has no supervised response tokens.
  - Never silently fall back to full-sequence loss.
- Add `type: sharegpt` and normalize deterministically:

```text
system             -> system
human, user        -> user
gpt, assistant     -> assistant
```

- Require string `from` and `value` fields.
- Reject unknown roles and non-string content.
- Do not use heuristic key inference as the public contract.
- Avoid logging full dataset URLs or sensitive query strings.

## Acceptance criteria

- `messages` with `loss: response` supervises assistant response tokens only.
- Omitted loss or `loss: all` retains full-sequence behavior.
- A valid one-row ShareGPT dataset normalizes successfully.
- ShareGPT records use the same rendering and loss pipeline as messages.
- Missing chat templates, unknown roles, malformed content, or conversations without assistant responses fail clearly.
- A fully masked dataset fails before `trainer.train()`.
- Existing Alpaca, prompt-completion, text, and full-sequence messages behavior remains unchanged.
- Documentation explains full-sequence versus response-only chat training.
- A one-step response-only GPU smoke run exports a valid GGUF.

## Tests

- Deterministic role-mapping tests, including a one-row ShareGPT regression.
- Tests for unknown roles, missing fields, empty conversations, and non-string content.
- Tests verifying `add_generation_prompt=False` and no Alpaca formatting.
- Missing/unusable chat-template tests.
- Label-level test proving user/prompt tokens are masked and response tokens remain supervised.
- Test proving a completely masked dataset fails.
- Exact-lock test with a tokenizer fixture containing known user/assistant markers.
- GPU fine-tune/export smoke test.

## Out of scope

- Native `assistant_only_loss`.
- User-provided Jinja templates or marker strings.
- Tool calls, `tool` messages, or multimodal content.
- Streaming datasets.
- Per-dataset loss settings.
- DPO, ORPO, KTO, GRPO, vision, or audio.
